subprojects/gradle/module-dependencies-graph/src/main/kotlin/com/avito/module/dependencies/ModuleConfigurationDependencies.kt

```diff
@@ -0,0 +1,22 @@
+package com.avito.module.dependencies
+
+import com.avito.module.configurations.ConfigurationType
+import org.gradle.api.Project
+import org.gradle.api.internal.artifacts.dependencies.DefaultProjectDependency
+
+public fun Project.dependenciesOnProjects(
+    types: Set<Class<out ConfigurationType>>
+): Set<DefaultProjectDependency> {
+    return configurations
+        .associateWith {
+            it.dependencies
+                .withType(DefaultProjectDependency::class.java)
+                // project has dependency to itself in a default configuration
+                .filter { it.dependencyProject != this }
+        }
+        .filterValues { it.isNotEmpty() }
+        .filterKeys { ConfigurationType.of(it).let { type -> types.any { it.isInstance(type) } } }
```

Comment:
Is it worth filtering configurations before getting dependencies in `associateWith`?

Reply:
Now operations the same but in the future determining of ConfigurationType will be heavier